dev/docker/Dockerfile

```diff
@@ -1,47 +1,49 @@
-FROM alpine:latest
-
-RUN apk add --no-cache gcc git make musl-dev go openssl curl
-
-# Configure Go
-ENV GOROOT /usr/lib/go
-ENV GOPATH /go
-ENV PATH /go/bin:$PATH
+# builder stage
+FROM golang as builder
+
+# define architectures which could be run rdpgw
+RUN dpkgArch="$(dpkg --print-architecture)"; \
+    case "$dpkgArch" in \
+        arm) ARCH='arm' ;; \
+        arm64) ARCH='arm64' ;; \
+        amd64) ARCH='amd64' ;; \
+        386) ARCH='386' ;; \
+        *) echo >&2 "error: unsupported architecture: $apkArch"; exit 1 ;; \
+    esac
 
-RUN random=$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 32 | head -n 1) && \
+# certificate
+RUN mkdir -p /opt/rdpgw && cd /opt/rdpgw && \
+    random=$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 32 | head -n 1) && \
     openssl genrsa -des3 -passout pass:$random -out server.pass.key 2048 && \
     openssl rsa -passin pass:$random -in server.pass.key -out key.pem && \
     rm server.pass.key && \
     openssl req -new -sha256 -key key.pem -out server.csr \
     -subj "/C=US/ST=VA/L=SomeCity/O=MyCompany/OU=MyDivision/CN=localhost" && \
     openssl x509 -req -days 365 -in server.csr -signkey key.pem -out server.pem
 
-RUN apkArch="$(apk --print-arch)"; \
-    case "$apkArch" in \
-        armhf) ARCH='arm' ;; \
-        aarch64) ARCH='arm64' ;; \
-        x86_64) ARCH='amd64' ;; \
-        x86) ARCH='386' ;; \ 
-        *) echo >&2 "error: unsupported architecture: $apkArch"; exit 1 ;; \
-    esac && \
-    git clone https://github.com/bolkedebruin/rdpgw.git && \
-    cd rdpgw && \
-    go mod tidy && \
-    go build -trimpath -tags '' -ldflags '' -o '/rdpgw/bin/rdpgw' ./cmd/rdpgw && \
-    mkdir -p /opt/rdpgw && \
-    mv /rdpgw/bin/rdpgw /opt/rdpgw/rdpgw && \
-    rm -rf /root/go && \
-    rm -rf /rdpgw
-
-RUN rm -rf /go
+# add user
+RUN adduser --disabled-password --gecos "" --home /opt/rdpgw --uid 1001 rdgw
```

Comment:
it night be smart to use `rdpgw` as a user as it is in line with the project name `rdgw` might create confusion

Reply:
fixed (sorry, I'm a little dyslexic)